Mage/src/main/java/mage/abilities/common/DrawCardTriggeredAbility.java

```diff
@@ -18,32 +18,35 @@
 /**
  * @author TheElk801
  */
-public class DrawSecondCardTriggeredAbility extends TriggeredAbilityImpl {
+public class DrawCardTriggeredAbility extends TriggeredAbilityImpl {
 
     private static final Hint hint = new ValueHint(
             "Cards drawn this turn", CardsDrawnThisTurnDynamicValue.instance
     );
     private final TargetController targetController;
+    private final int cardNumber;
 
-    public DrawSecondCardTriggeredAbility(Effect effect, boolean optional) {
-        this(effect, optional, TargetController.YOU);
+    public DrawCardTriggeredAbility(Effect effect, boolean optional, int cardNumber) {
+        this(effect, optional, TargetController.YOU, cardNumber);
     }
 
-    public DrawSecondCardTriggeredAbility(Effect effect, boolean optional, TargetController targetController) {
-        this(Zone.BATTLEFIELD, effect, optional, targetController);
+    public DrawCardTriggeredAbility(Effect effect, boolean optional, TargetController targetController, int cardNumber) {
+        this(Zone.BATTLEFIELD, effect, optional, targetController, cardNumber);
     }
 
-    public DrawSecondCardTriggeredAbility(Zone zone, Effect effect, boolean optional, TargetController targetController) {
+    public DrawCardTriggeredAbility(Zone zone, Effect effect, boolean optional, TargetController targetController, int cardNumber) {
         super(zone, effect, optional);
-        this.addWatcher(new DrawSecondCardWatcher());
+        this.addWatcher(new DrawCardWatcher());
```

Comment:
@theelk801 why can't we store the reference to the card watcher ?
so we can avoid using it staticly ?

Reply:
ok, this will stay the way it is